src/deroff.rs

```diff
@@ -1168,6 +1155,308 @@ fn deroff_files(files: &[String]) -> io::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_text_arg() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Hello World!".into();
+    assert!(deroffer.text_arg());
+    assert_eq!(deroffer.s, " World!");
+    assert_eq!(deroffer.output.take(), "Hello");
+
+    let mut deroffer = Deroffer::new();
+    assert!(!deroffer.text_arg());
+    assert!(deroffer.s.is_empty());
+    assert!(deroffer.output.take().is_empty());
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "\t\n\t           \t\n".into();
+    assert!(!deroffer.text_arg());
+    assert_eq!(deroffer.s, "\t\n\t           \t\n");
+    assert!(deroffer.output.take().is_empty());
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = r"\r".into();
+    assert!(!deroffer.text_arg());
+    assert!(deroffer.s.is_empty());
+    assert_eq!(deroffer.output.take(), "r");
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Applebees".into();
+    assert!(deroffer.text_arg());
+    assert_eq!(deroffer.s, "");
+    assert_eq!(deroffer.output.take(), "Applebees");
```

Comment:
```suggestion
    assert_eq!(deroffer.output.take(), "Applebees");
    
    let mut deroffer = Deroffer::new();
    deroffer.s = "忍一时风平浪静，退一步海口天空。".into();
    assert!(deroffer.text_arg());
    assert_eq!(deroffer.s, "忍一时风平浪静，退一步海口天空。");
    assert_eq!(deroffer.output.take(), "");
```

Oh, ours doesn't work on CJK characters. The original works.

```python
>>> d = Deroffer()
>>> d.s = "忍一时风平浪静，退一步海口天空。"
>>> d.text_arg()
True
>>> d.output
['忍一时风平浪静，退一步海口天空。']
>>> d.s
''
```